src/Rector/Dynamic/FluentReplaceRector.php

```diff
@@ -0,0 +1,31 @@
+<?php declare(strict_types=1);
+
+namespace Rector\Rector\Dynamic;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Return_;
+use PhpParser\Node\Expr\Variable;
+use Rector\Rector\AbstractRector;
+
+final class FluentReplaceRector extends AbstractRector
+{
+    public function isCandidate(Node $node): bool
+    {
+        if (! $node instanceof Return_) {
+            return false;
+        }
+
+        $returnExpr = $node->expr;
+
+        if (! $returnExpr instanceof Variable) {
+            return false;
+        }
+
+        return $returnExpr->name === 'this';
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        return null;
```

Comment:
@TomasVotruba Returning `null` shouldn't remove `return $this`?

Reply:
It should be `Nop` instead. Null means *do nothing*

Reply:
I keep receiving an error: `Uninitialized string offset: 0` :disappointed: